src/registries/auto_completes/auto_complete_registry.ts

```diff
@@ -36,6 +36,7 @@ export interface AutoCompleteProviderDefinition {
   sequence?: number;
   autoSelectFirstProposal?: boolean;
   maxDisplayedProposals?: number;
+  keepOpenOnExactMatch?: boolean; // By default, the auto complete closes when the exact match is found.
```

Comment:
I'm unsure we want to go in this direction, because I don't think that functionnaly we ever want to close the provider if we have an exact match ? I don't think it's something function specific.

For example if you have a pivot with id 1 and id 11, why would typing 1 remove the option 11 ? Same for data validation and other autocompleted stuff.

I don't think the exact match check make much sense. Maybe if we have an exact match AND there is no other matches we could close the automplete. But why bother? 

Looking at the original comment `this means the user has chosen a proposal` maybe the goal was to close the autocomplete when the use clicked on a proposal. In functions, it's done automatically because selecting a proposal also add a parenthesis. We could probably have a similar bheviour for the otehr autocomplete (autocompleting a pivot field would add a comma for example). 

@LucasLefevre what do you think ?

Reply:
I agree with Adrien, it's not specific to functions.

> Maybe if we have an exact match AND there is no other matches we could close the automplete

close it...or display everything ? It looks like it's the behavior of the data validation dropdown list, but we can change the behavior to generalize and keep it simple.

Reply:
So @dhrp-odoo I think you can drop this task, I don't think it's a simple fix, we'll need to discuss it in more detail with the PO and experiment a bit to get a behaviour that fit all the use cases

Reply:
Hello @hokolomopo @LucasLefevre 👋

Sure, I've been experimenting with this since this morning and have identified some edge cases that need to be addressed:

- When typing =SUM, the composer assistant provides some values. However, if our intention is to write =SUMIFS, upon exact match, we receive an odd assistant suggestion. This is likely due to how we handle data validation assistant.
- Typing something unusual like =SUSUBUSBD also yields strange results, may be similar to the issue above.
- When typing =PIVOT.VALUE(1,"Expected Revenue","Stage", "Ne"), we are not getting any suggestions.

Reply:
Nice, thanks!
I added your findings in the task to make sure they are addressed